### PR TITLE
[depends] gnutls 3.6.14 and nettle 3.6 bump

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Security fixes in Gnutls 3.6.14 required. Will open backport PR shortly.

## Motivation and Context

Security issues in gnutls.

## How Has This Been Tested?

Mac

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: tools/depends/target/gnutls/add-dl-as-private-lib.patch [hunk 1]
```diff
@@ -1,22 +1,24 @@
---- a/configure.ac
-+++ b/configure.ac
-@@ -512,6 +512,9 @@
- 
- 
+diff -ur gnutls/configure.ac gnutls-patched/configure.ac
```
Comment (Rechi) [inline tools/depends/target/gnutls/add-dl-as-private-lib.patch:1]:
```suggestion
```

File: tools/depends/target/gnutls/add-dl-as-private-lib.patch [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
+--- gnutls/configure.ac	2020-06-03 14:00:25.000000000 +0100
```
Comment (Rechi) [inline tools/depends/target/gnutls/add-dl-as-private-lib.patch:2]:
```suggestion
--- a/configure.ac
```

File: tools/depends/target/gnutls/add-dl-as-private-lib.patch [hunk 3]
```diff
@@ -1,22 +1,24 @@
...
++++ gnutls-patched/configure.ac	2020-07-16 09:56:53.000000000 +0100
```
Comment (Rechi) [inline tools/depends/target/gnutls/add-dl-as-private-lib.patch:3]:
```suggestion
+++ b/configure.ac
```

File: tools/depends/target/gnutls/add-dl-as-private-lib.patch [hunk 4]
```diff
@@ -1,22 +1,24 @@
...
-+      AC_SUBST([LIBDL], [-ldl])
++  AC_SUBST([LIBDL], [-ldl])
 +fi
- 
+
  AC_ARG_ENABLE(fips140-mode,
    AS_HELP_STRING([--enable-fips140-mode], [enable FIPS140-2 mode]),
---- a/lib/gnutls.pc.in
-+++ b/lib/gnutls.pc.in
+diff -ur gnutls/lib/gnutls.pc.in gnutls-patched/lib/gnutls.pc.in
```
Comment (Rechi) [inline tools/depends/target/gnutls/add-dl-as-private-lib.patch:14]:
```suggestion
```

File: tools/depends/target/gnutls/add-dl-as-private-lib.patch [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
+--- gnutls/lib/gnutls.pc.in	2020-06-02 05:09:50.000000000 +0100
```
Comment (Rechi) [inline tools/depends/target/gnutls/add-dl-as-private-lib.patch:15]:
```suggestion
--- a/lib/gnutls.pc.in
```

File: tools/depends/target/gnutls/add-dl-as-private-lib.patch [hunk 6]
```diff
@@ -1,22 +1,24 @@
...
++++ gnutls-patched/lib/gnutls.pc.in	2020-07-16 09:57:30.000000000 +0100
```
Comment (Rechi) [inline tools/depends/target/gnutls/add-dl-as-private-lib.patch:16]:
```suggestion
+++ b/lib/gnutls.pc.in
```

**Review by Rechi — COMMENTED** [review]
Please follow the common patch style used for all depends.

File: tools/depends/target/gnutls/add-dl-as-private-lib.patch [hunk 7]
```diff
@@ -16,7 +16,7 @@
  URL: https://www.gnutls.org/
  Version: @VERSION@
  Libs: -L${libdir} -lgnutls
--Libs.private: @LIBINTL@ @LIBSOCKET@ @INET_PTON_LIB@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBIDN2_LIBS@ @LIBATOMIC_LIBS@
-+Libs.private: @LIBINTL@ @LIBSOCKET@ @INET_PTON_LIB@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBIDN2_LIBS@ @LIBATOMIC_LIBS@ @LIBDL@
+-Libs.private: @LIBINTL@ @LIBSOCKET@ @INET_PTON_LIB@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBATOMIC_LIBS@ @LIB_CRYPT32@ @LIBNCRYPT@ @LIBBCRYPT@
++Libs.private: @LIBINTL@ @LIBSOCKET@ @INET_PTON_LIB@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBATOMIC_LIBS@ @LIB_CRYPT32@ @LIBNCRYPT@ @LIBBCRYPT@ @LIBDL@
```
Comment (fuzzard) [inline tools/depends/target/gnutls/add-dl-as-private-lib.patch:20]: Im guessing these are a carryover from ffmpegdirect where you are using it for windows. do we actually require the crypto libs to be added for this? Keep in mind, these patches are NOT used for windows at all.

Comment (fuzzard) [reply tools/depends/target/gnutls/add-dl-as-private-lib.patch:20]: Also, libidn can be explicitly disabled with ----without-idn, so removing the LIBIDN2_LIBS should not be necessary

Comment (phunkyfish) [reply tools/depends/target/gnutls/add-dl-as-private-lib.patch:20]: This is the pc file shipped with 3.6.14. All I did was add LIBDL to the end of the supplies file similar to the previous versions patch.

**Comment by DaveTBlake** [comment]
Thanks @phunkyfish 
@wsnipex happy with this since you raised the issue?

**Review by wsnipex — APPROVED** [review]

**Pull request merged**